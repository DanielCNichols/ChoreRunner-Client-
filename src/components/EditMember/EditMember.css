/* div.valid-error {
  color: white;
  font-weight: 600;
} */

.edit-member-form {
  box-sizing: border-box;
    margin: 10px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    /* padding: 10px 20px; */
    border: none;
}

.edit-member-form fieldset {
  border: none;
    margin: 0 auto;
    padding: 0 10px;
    width: 90%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.edit-member-form input {
  margin: 0 auto;
  width: 80%;
}

.edit-member-form label {
  font-size: 20px;
  color: rgb(255, 217, 0);
  padding-bottom: 5px;
  font-family: 'Electrolize', sans-serif;
  margin: 10px auto 5px auto;
  width: 80%;
}



.edit-member-form button {
  text-align: center;
  margin: 20px auto;
  font-family: 'Electrolize', sans-serif;
  font-size: 20px;
  padding: 10px;
  background-color: rgb(255, 217, 0);
  text-shadow: -1px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725f00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 0.3em rgb(134, 112, 0);
  cursor: pointer;
}

.edit-member-form legend {
  font-family: 'Press Start 2P', cursive;
  color: rgb(255, 217, 0);
  line-height: 25px;
  margin: 5px auto;
  text-align: center;
}

.alert p {
  font-family: 'Electrolize', sans-serif;
  color: white;
  max-width: 245px;
  word-wrap: break-word;
}

.alert button {
  text-align: center;
  margin: 20px auto;
  font-family: 'Electrolize', sans-serif;
  font-size: 20px;
  padding: 10px;
  background-color: rgb(255, 217, 0);
  text-shadow: -1px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725f00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 0.3em rgb(134, 112, 0);
  cursor: pointer;
}