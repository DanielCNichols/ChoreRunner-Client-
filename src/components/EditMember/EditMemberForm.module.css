@import '../../vars.css';

.error {
  color: red;
  font-weight: bold;
  margin-bottom: 5px;
}

.editMemberForm {
  width: 100%;
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border: 2px solid white;
  border-radius: 6px;
  background-color: var(--color-menus);
}

.editMemberForm fieldset {
  min-width: 245px;
  border: none;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.editMemberForm legend {
  text-align: center;
  font-size: 1.2em;
}
.editMemberForm legend,
.formElement {
  margin-bottom: 20px;
}

.formElement label {
  text-align: left;
  margin-bottom: 5px;
}

.formButtons {
  width: 100%;
  display: flex;
  justify-content: space-between;
  margin: 10px auto;
}

.formButtons button {
  width: 100px;
  height: 45px;
  text-align: center;
  margin: 10px auto;
  font-family: 'Press Start 2P', cursive;
  font-size: 0.8em;
  padding: 7px;
  background-color: rgb(255, 217, 0);
  text-shadow: -1px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725f00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 0.3em rgb(134, 112, 0);
  cursor: pointer;
}
