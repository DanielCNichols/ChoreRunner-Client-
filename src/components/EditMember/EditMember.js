import React from 'react';
import ApiService from '../../services/api-service';
//import HouseHoldContext from '../../contexts/HouseHoldContext'
import './EditMember.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPencilAlt } from '@fortawesome/free-solid-svg-icons';

export default class EditMember extends React.Component {
  constructor(props) {
    console.log(props.member);
    let { id, name, username, password } = props.member;
    super(props);
    this.state = {
      id,
      name,
      username,
      password,
      touched: false,
      error: null,
      validateError: {
        nameError: '',
        usernameError: '',
        passwordError: '',
      },
    };
  }

  onChangeHandle = e => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  toggleTouched = () => {
    this.setState({touched: true})
  }

  validate = (name, username, pass) => {
    let nameError = '';
    let usernameError = '';
    let passwordError ='';

    if (this.state.name.length <= 2) {
      nameError = 'Please enter 3 characters or more';
    }

    // Validates child's username
    if (username.length > 50) {
      usernameError = 'Your name must be less than 50 characters';
    }

    //validate the password
    if (pass) {
      if (pass.length > 10 || pass.length < 4) {
      passwordError = 'Your password must be more than 4 and less than 40 characters'
    }
  }


    if (nameError || usernameError || passwordError) {
      this.setState({ validateError: { usernameError, nameError, passwordError } });
      return false;
    }
    return true;
  };

  handleSubmit = e => {
    e.preventDefault();
    let {householdId} = this.props
    
    let updatedMember = {
      id: this.state.id,
      name: this.state.name.trim(),
      username: this.state.username.trim(),
      password: this.state.touched ? this.state.password.trim() : null
    };
    
    let isValid = this.validate(updatedMember.name, updatedMember.username, updatedMember.password);
    console.log('This is the password being sent', updatedMember.password)

    if (isValid) {
      ApiService.editMember(updatedMember, householdId, this.state.id)
        .then(res => {
          //This.handleSomething
          this.props.editMember(res)
        })
        .catch(error => this.setState({ error: error.error }));
    }
  };


  render() {
    const { usernameError, nameError, passwordError } = this.state.validateError;
    const { error, name, username} = this.state;
    return (
      <form onSubmit={this.handleSubmit} id="edit-member-form">
        <fieldset>
          <legend>Edit Member</legend>
          <label htmlFor="member-name">Name</label>
          <input
            type="text"
            name="name"
            value={name}
            onChange={this.onChangeHandle}
          ></input>
          <label htmlFor="child-username">Child username</label>
          <input
            type="text"
            name="username"
            value={username}
            onChange={this.onChangeHandle}
          ></input>
          <label htmlFor="child-password">Child password</label>
          <input
            type="password"
            name="password"
            onFocus={this.toggleTouched}
            onChange={this.onChangeHandle}
          ></input>
          <div className="valid-error">
            {nameError || usernameError || passwordError || error}
          </div>
          <button type="submit" className="submit-edit-member">
            Submit Changes
          </button>
          <button type="button" onClick={this.props.handleCancel} className="cancel">
            Cancel
          </button>
        </fieldset>
      </form>
    );
  }
}
