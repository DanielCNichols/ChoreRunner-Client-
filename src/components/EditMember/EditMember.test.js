import React from 'react';
import ReactDOM from 'react-dom';
import EditMember from './EditMember';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('EditMember component testing', () => {
  const member = {
    member_id: 1,
    name: 'test',
    username: 'test-user',
    household_id: 2,
    password: 'pass',
  };

  const editMember = () => {};
  const toggleEdit = () => {};

  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <MemoryRouter>
        <EditMember
          member={member}
          editMember={editMember}
          toggleEdit={toggleEdit}
        />
      </MemoryRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <EditMember
            member={member}
            editMember={editMember}
            toggleEdit={toggleEdit}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
