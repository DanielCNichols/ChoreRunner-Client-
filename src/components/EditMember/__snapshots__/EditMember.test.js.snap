// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMember component testing renders UI as expected 1`] = `
<form
  className="editMemberForm"
  onSubmit={[Function]}
>
  <fieldset
    className="Fieldset"
  >
    <legend
      className="Legend videoGameTitles"
    >
      Edit Member
    </legend>
    <div
      className="FormElement formElement"
    >
      <label
        className="Label"
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="Input"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="Elliott"
        type="text"
        value="test"
      />
    </div>
    <div
      className="FormElement formElement"
    >
      <label
        className="Label"
        htmlFor="username"
      >
        Member Username
      </label>
      <input
        className="Input"
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="elliot123"
        type="text"
        value="test-user"
      />
    </div>
    <div
      className="FormElement formElement"
    >
      <label
        className="Label"
        htmlFor="password"
      >
        Member Password
      </label>
      <input
        className="Input"
        id="password"
        name="password"
        onChange={[Function]}
        onFocus={[Function]}
        type="password"
      />
    </div>
    <div
      className="FormElement formElement"
    >
      <label
        className="Label"
        htmlFor="confirmPass"
      >
        Confirm Password
      </label>
      <input
        className="Input"
        id="confirmPass"
        name="confirmPass"
        onChange={[Function]}
        type="password"
      />
    </div>
    <div
      className="formButtons"
    >
      <button
        className="arcadeButton"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="arcadeButton"
        type="submit"
      >
        Submit
      </button>
    </div>
  </fieldset>
</form>
`;
